pandas/tests/io/test_rdata.py

```diff
@@ -0,0 +1,565 @@
+from io import BytesIO
+import os
+from urllib.error import HTTPError
+
+import pytest
+
+import pandas.util._test_decorators as td
+
+from pandas import (
+    DataFrame,
+    Timestamp,
+)
+import pandas._testing as tm
+
+from pandas.io.rdata._rdata import (
+    LibrdataParserError,
+    LibrdataWriterError,
+)
+from pandas.io.rdata.rdata_reader import read_rdata
+
+ghg_df = DataFrame(
+    {
+        "gas": {
+            141: "Carbon dioxide",
+            142: "Methane",
+            143: "Nitrous oxide",
+            144: "Fluorinated gases",
+            145: "Total",
+        },
+        "year": {141: 2018, 142: 2018, 143: 2018, 144: 2018, 145: 2018},
+        "emissions": {
+            141: 5424.881502132882,
+            142: 634.4571270782675,
+            143: 434.52855537666636,
+            144: 182.78243246177678,
+            145: 6676.649617049592,
+        },
+    }
+).rename_axis("rownames")
+
+plants_df = DataFrame(
+    {
+        "plant_group": {
+            16: "Pteridophytes",
+            17: "Pteridophytes",
+            18: "Pteridophytes",
+            19: "Pteridophytes",
+            20: "Pteridophytes",
+        },
+        "status": {
+            16: "Data Deficient",
+            17: "Extinct",
+            18: "Not Threatened",
+            19: "Possibly Threatened",
+            20: "Threatened",
+        },
+        "count": {16: 398, 17: 65, 18: 1294, 19: 408, 20: 1275},
+    }
+).rename_axis("rownames")
+
+sea_ice_df = DataFrame(
+    {
+        "year": {1012: 2016, 1013: 2017, 1014: 2018, 1015: 2019, 1016: 2020},
+        "mo": {1012: 12, 1013: 12, 1014: 12, 1015: 12, 1016: 12},
+        "data.type": {
+            1012: "Goddard",
+            1013: "Goddard",
+            1014: "Goddard",
+            1015: "Goddard",
+            1016: "NRTSI-G",
+        },
+        "region": {1012: "S", 1013: "S", 1014: "S", 1015: "S", 1016: "S"},
+        "extent": {1012: 8.28, 1013: 9.48, 1014: 9.19, 1015: 9.41, 1016: 10.44},
+        "area": {1012: 5.51, 1013: 6.23, 1014: 5.59, 1015: 6.59, 1016: 6.5},
+    }
+).rename_axis("rownames")
+
+ppm_df = DataFrame(
+    {
+        "date": {
```

Comment:
Can R data files have dates outside the date range supported by pandas?  What happens there? Can you add a test for this case?

Reply:
What date range would that be?

Reply:
Pandas only supports dates in +/- 293 years from 1970.
```
In [29]: pd.Timestamp.min
Out[29]: Timestamp('1677-09-21 00:12:43.145225')

In [30]: pd.Timestamp.max
Out[30]: Timestamp('2262-04-11 23:47:16.854775807')
```

Reply:
A related question - are there any forbidden names for objects in rdata files?  For example, keywords in R?

Reply:
R has a handful [reserved words](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Reserved.html) that are not allowed for object names unless escaped with backticks.